DESIGN.md

```diff
@@ -298,8 +298,9 @@ type MeshCataloger interface {
 	// ExpectProxy catalogs the fact that a certificate was issued for an Envoy proxy and this is expected to connect to XDS.
 	ExpectProxy(certificate.CommonName)
 
-	// GetServiceFromEnvoyCertificate returns the single service given Envoy is a member of based on the certificate provided, which is a cert issued to an Envoy for XDS communication (not Envoy-to-Envoy).
-	GetServiceFromEnvoyCertificate(certificate.CommonName) (*service.MeshService, error)
+    // GetServicesFromEnvoyCertificate returns a list of services the given Envoy is a member of based on the certificate provided, 
+    // which is a cert issued to an Envoy for XDS communication (not Envoy-to-Envoy).
+	GetServicesFromEnvoyCertificate(certificate.CommonName) ([]*service.MeshService, error)
```

Comment:
@ksubrmnn We used `*service.MeshService` here so we could return `nil` when needed.  But now that we have slices - we can return an empty slice.
Slices of pointers to something would be useful when we need some indices in the slice to be nil, but I don't see us needing this.  Would you be so kind to change this from `[]*service.MeshService` to `[]service.MeshService`.
Here is an interesting article on the subject:

Reply:
Thanks for explaining the original reasoning!